rspamd.service

```diff
@@ -9,9 +9,39 @@ NonBlocking=true
 ExecStart=/usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
 ExecReload=/bin/kill -HUP $MAINPID
 User=_rspamd
+ConfigurationDirectory=rspamd
 RuntimeDirectory=rspamd
 RuntimeDirectoryMode=0755
+StateDirectory=rspamd
+StateDirectoryMode=0700
+LogsDirectory=rspamd
 Restart=always
 
+AmbientCapabilities=
+CapabilityBoundingSet=
+DevicePolicy=closed
+LockPersonality=true
+NoNewPrivileges=true
+PrivateDevices=true
+PrivateMounts=true
+PrivateTmp=true
+PrivateUsers=true
+ProtectClock=true
+ProtectControlGroups=true
+ProtectHome=true
+ProtectHostname=true
+ProtectKernelLogs=true
+ProtectKernelModules=true
+ProtectKernelTunables=true
+ProtectSystem=strict
+RemoveIPC=true
+RestrictAddressFamilies=AF_INET AF_INET6 AF_UNIX
+RestrictNamespaces=true
+RestrictRealtime=true
+RestrictSUIDSGID=true
+SystemCallArchitectures=native
+SystemCallFilter=@system-service
```

Comment:
Full list of all allowed system calls:

- "io_cancel\0"
- "io_destroy\0"
- "io_getevents\0"
- "io_pgetevents\0"
- "io_pgetevents_time64\0"
- "io_setup\0"
- "io_submit\0"
- "io_uring_enter\0"
- "io_uring_register\0"
- "io_uring_setup\0"
- "_llseek\0"
- "close\0"
- "dup\0"
- "dup2\0"
- "dup3\0"
- "lseek\0"
- "pread64\0"
- "preadv\0"
- "preadv2\0"
- "pwrite64\0"
- "pwritev\0"
- "pwritev2\0"
- "read\0"
- "readv\0"
- "write\0"
- "writev\0"
- "chown\0"
- "chown32\0"
- "fchown\0"
- "fchown32\0"
- "fchownat\0"
- "lchown\0"
- "lchown32\0"
- "clock_getres\0"
- "clock_getres_time64\0"
- "clock_gettime\0"
- "clock_gettime64\0"
- "clock_nanosleep\0"
- "clock_nanosleep_time64\0"
- "execve\0"
- "exit\0"
- "exit_group\0"
- "futex\0"
- "futex_time64\0"
- "get_robust_list\0"
- "get_thread_area\0"
- "getegid\0"
- "getegid32\0"
- "geteuid\0"
- "geteuid32\0"
- "getgid\0"
- "getgid32\0"
- "getgroups\0"
- "getgroups32\0"
- "getpgid\0"
- "getpgrp\0"
- "getpid\0"
- "getppid\0"
- "getresgid\0"
- "getresgid32\0"
- "getresuid\0"
- "getresuid32\0"
- "getrlimit\0"      /* make sure processes can query stack size and such */
- "getsid\0"
- "gettid\0"
- "gettimeofday\0"
- "getuid\0"
- "getuid32\0"
- "membarrier\0"
- "nanosleep\0"
- "pause\0"
- "prlimit64\0"
- "restart_syscall\0"
- "rseq\0"
- "rt_sigreturn\0"
- "sched_yield\0"
- "set_robust_list\0"
- "set_thread_area\0"
- "set_tid_address\0"
- "set_tls\0"
- "sigreturn\0"
- "time\0"
- "ugetrlimit\0"
- "access\0"
- "chdir\0"
- "chmod\0"
- "close\0"
- "creat\0"
- "faccessat\0"
- "fallocate\0"
- "fchdir\0"
- "fchmod\0"
- "fchmodat\0"
- "fcntl\0"
- "fcntl64\0"
- "fgetxattr\0"
- "flistxattr\0"
- "fremovexattr\0"
- "fsetxattr\0"
- "fstat\0"
- "fstat64\0"
- "fstatat64\0"
- "fstatfs\0"
- "fstatfs64\0"
- "ftruncate\0"
- "ftruncate64\0"
- "futimesat\0"
- "getcwd\0"
- "getdents\0"
- "getdents64\0"
- "getxattr\0"
- "inotify_add_watch\0"
- "inotify_init\0"
- "inotify_init1\0"
- "inotify_rm_watch\0"
- "lgetxattr\0"
- "link\0"
- "linkat\0"
- "listxattr\0"
- "llistxattr\0"
- "lremovexattr\0"
- "lsetxattr\0"
- "lstat\0"
- "lstat64\0"
- "mkdir\0"
- "mkdirat\0"
- "mknod\0"
- "mknodat\0"
- "mmap\0"
- "mmap2\0"
- "munmap\0"
- "newfstatat\0"
- "oldfstat\0"
- "oldlstat\0"
- "oldstat\0"
- "open\0"
- "openat\0"
- "readlink\0"
- "readlinkat\0"
- "removexattr\0"
- "rename\0"
- "renameat\0"
- "renameat2\0"
- "rmdir\0"
- "setxattr\0"
- "stat\0"
- "stat64\0"
- "statfs\0"
- "statfs64\0"
-#x
- "statx\0"
-#
- "symlink\0"
- "symlinkat\0"
- "truncate\0"
- "truncate64\0"
- "unlink\0"
- "unlinkat\0"
- "utime\0"
- "utimensat\0"
- "utimensat_time64\0"
- "utimes\0"
- "_newselect\0"
- "epoll_create\0"
- "epoll_create1\0"
- "epoll_ctl\0"
- "epoll_ctl_old\0"
- "epoll_pwait\0"
- "epoll_wait\0"
- "epoll_wait_old\0"
- "eventfd\0"
- "eventfd2\0"
- "poll\0"
- "ppoll\0"
- "ppoll_time64\0"
- "pselect6\0"
- "pselect6_time64\0"
- "select\0"
- "ipc\0"
- "memfd_create\0"
- "mq_getsetattr\0"
- "mq_notify\0"
- "mq_open\0"
- "mq_timedreceive\0"
- "mq_timedreceive_time64\0"
- "mq_timedsend\0"
- "mq_timedsend_time64\0"
- "mq_unlink\0"
- "msgctl\0"
- "msgget\0"
- "msgrcv\0"
- "msgsnd\0"
- "pipe\0"
- "pipe2\0"
- "process_vm_readv\0"
- "process_vm_writev\0"
- "semctl\0"
- "semget\0"
- "semop\0"
- "semtimedop\0"
- "semtimedop_time64\0"
- "shmat\0"
- "shmctl\0"
- "shmdt\0"
- "shmget\0"
- "add_key\0"
- "keyctl\0"
- "request_key\0"
- "mlock\0"
- "mlock2\0"
- "mlockall\0"
- "munlock\0"
- "munlockall\0"
- "accept\0"
- "accept4\0"
- "bind\0"
- "connect\0"
- "getpeername\0"
- "getsockname\0"
- "getsockopt\0"
- "listen\0"
- "recv\0"
- "recvfrom\0"
- "recvmmsg\0"
- "recvmmsg_time64\0"
- "recvmsg\0"
- "send\0"
- "sendmmsg\0"
- "sendmsg\0"
- "sendto\0"
- "setsockopt\0"
- "shutdown\0"
- "socket\0"
- "socketcall\0"
- "socketpair\0"
- "arch_prctl\0"
- "capget\0"      /* Able to query arbitrary processes */
- "clone\0"
- "clone3\0"
- "execveat\0"
- "fork\0"
- "getrusage\0"
- "kill\0"
- "pidfd_open\0"
- "pidfd_send_signal\0"
- "prctl\0"
- "rt_sigqueueinfo\0"
- "rt_tgsigqueueinfo\0"
- "setns\0"
- "swapcontext\0" /* Some archs e.g. powerpc32 are using it to do userspace context switches */
- "tgkill\0"
- "times\0"
- "tkill\0"
- "unshare\0"
- "vfork\0"
- "wait4\0"
- "waitid\0"
- "waitpid\0"
- "ioprio_set\0"
- "mbind\0"
- "migrate_pages\0"
- "move_pages\0"
- "nice\0"
- "sched_setaffinity\0"
- "sched_setattr\0"
- "sched_setparam\0"
- "sched_setscheduler\0"
- "set_mempolicy\0"
- "setpriority\0"
- "setrlimit\0"
- "setgid\0"
- "setgid32\0"
- "setgroups\0"
- "setgroups32\0"
- "setregid\0"
- "setregid32\0"
- "setresgid\0"
- "setresgid32\0"
- "setresuid\0"
- "setresuid32\0"
- "setreuid\0"
- "setreuid32\0"
- "setuid\0"
- "setuid32\0"
- "rt_sigaction\0"
- "rt_sigpending\0"
- "rt_sigprocmask\0"
- "rt_sigsuspend\0"
- "rt_sigtimedwait\0"
- "rt_sigtimedwait_time64\0"
- "sigaction\0"
- "sigaltstack\0"
- "signal\0"
- "signalfd\0"
- "signalfd4\0"
- "sigpending\0"
- "sigprocmask\0"
- "sigsuspend\0"
- "fdatasync\0"
- "fsync\0"
- "msync\0"
- "sync\0"
- "sync_file_range\0"
- "sync_file_range2\0"
- "syncfs\0"
- "alarm\0"
- "getitimer\0"
- "setitimer\0"
- "timer_create\0"
- "timer_delete\0"
- "timer_getoverrun\0"
- "timer_gettime\0"
- "timer_gettime64\0"
- "timer_settime\0"
- "timer_settime64\0"
- "timerfd_create\0"
- "timerfd_gettime\0"
- "timerfd_gettime64\0"
- "timerfd_settime\0"
- "timerfd_settime64\0"
- "times\0"
- "brk\0"
- "capget\0"
- "capset\0"
- "copy_file_range\0"
- "fadvise64\0"
- "fadvise64_64\0"
- "flock\0"
- "get_mempolicy\0"
- "getcpu\0"
- "getpriority\0"
- "getrandom\0"
- "ioctl\0"
- "ioprio_get\0"
- "kcmp\0"
- "madvise\0"
- "mprotect\0"
- "mremap\0"
- "name_to_handle_at\0"
- "oldolduname\0"
- "olduname\0"
- "personality\0"
- "readahead\0"
- "readdir\0"
- "remap_file_pages\0"
- "sched_get_priority_max\0"
- "sched_get_priority_min\0"
- "sched_getaffinity\0"
- "sched_getattr\0"
- "sched_getparam\0"
- "sched_getscheduler\0"
- "sched_rr_get_interval\0"
- "sched_rr_get_interval_time64\0"
- "sched_yield\0"
- "sendfile\0"
- "sendfile64\0"
- "setfsgid\0"
- "setfsgid32\0"
- "setfsuid\0"
- "setfsuid32\0"
- "setpgid\0"
- "setsid\0"
- "splice\0"
- "sysinfo\0"
- "tee\0"
- "umask\0"
- "uname\0"
- "userfaultfd\0"
- "vmsplice\0"